namespace-registry/client_commands.go

```diff
@@ -0,0 +1,188 @@
+package nsregistry
+
+import (
+	"crypto/tls"
+
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"net/http"
+	"bytes"
+	"bufio"
+	"math/big"
+	"encoding/base64"
+
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/spf13/viper"
+)
+
+func makeRequest(url string, method string, data map[string]interface{}) ([]byte, error) {
+	payload, _ := json.Marshal(data)
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	if viper.GetBool("TLSSkipVerify") {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		client = &http.Client{Transport: tr}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil,err
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	return body, nil
```

Comment:
This function doesn't check the HTTP status code, meaning a `500`- or `403`-type response is going to be marked as successful.